pyccel/ast/core.py

```diff
@@ -4593,7 +4593,13 @@ def __init__(
     def __getitem__(self, *args):
 
         if self.shape and len(self.shape) != len(args):
-            raise IndexError('Rank mismatch.')
+
+            raise IndexException('Rank mismatch.')
```

Comment:
Why `IndexException` instead of `IndexError`? This gives me:
```
NameError: name 'IndexException' is not defined
```

Reply:
I've just rolled it back to `IndexError`